File: CreationalPattern/FactoryPattern/Golang/Plants/SunFlower.go

package Plants


/**
 * SunFlower struct, implement Plant interface
 */
type SunFlower struct {
	Name  string
	Price int
	// TODO add more attribute ...
}

// Constructor for SunFlower 
func NewSunFlower(params...interface{}) *SunFlower {
	p := &SunFlower {
		Name: "SunFlower",
		Price: 50,
	}

	len := len(params)

	if (len > 0) {
		p.Name = params[0].(string)
	}

	if (len > 1) {
		p.Price = params[1].(int)
	}

	return p
}

func (s *SunFlower) PlantName() string {
	return s.Name
}

func (s *SunFlower) PlantPrice() int {
	return s.Price
}

File: CreationalPattern/AbstractFactoryPattern/Java/Plants/SunFlower.java

package Plants;

import Interface.Plant;

/**
 * Class for Sunflower
 */
public class SunFlower implements Plant {
    // Var params for Sunflower
    private String name;
    private int hp;
    private int price;
    
    // Constructor
    public SunFlower() {
        this.name = "Sunflower";
        this.hp = 100;
        this.price = 50;
    }

    public SunFlower(String name, int hp, int price) {
        this.name = name;
        this.hp = hp;
        this.price = price;
    }
    
    // Override method Display from Plant
    @Override
    public void display() {
        System.out.println("Name: " + name + ", Price: " + price + ", HP: " + hp);
    }

    // Override method GetHealth from Plant
    @Override
    public int getHealth() {
        return hp;
    }

    // Getter and setter
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getPrice() {
        return price;
    }

    public void setPrice(int price) {
        this.price = price;
    }

    public void setHp(int hp) {
        this.hp = hp;
    }

    @Override
    public void show() {
        // TODO Auto-generated method stub
        throw new UnsupportedOperationException("Unimplemented method 'show'");
    }
}